pkg/operator/staticpod/controller/guard/guard_controller.go

```diff
@@ -181,25 +196,24 @@ func (c *GuardController) sync(ctx context.Context, syncCtx factory.SyncContext)
 			pdb.Spec.MinAvailable = &minAvailable
 		}
 
-		// List the pdb from the cache in case it exists and there's nothing to update
-		// so no Get request is executed.
-		pdbs, err := c.pdbLister.PodDisruptionBudgets(c.targetNamespace).List(labels.Everything())
-		if err != nil {
-			klog.Errorf("Unable to list PodDisruptionBudgets: %v", err)
-			return err
-		}
-
-		for _, pdbItem := range pdbs {
-			if pdbItem.Name == pdb.Name {
-				if pdbItem.Spec.MinAvailable != pdb.Spec.MinAvailable {
-					_, _, err = resourceapply.ApplyPodDisruptionBudget(ctx, c.pdbGetter, syncCtx.Recorder(), pdb)
-					if err != nil {
-						klog.Errorf("Unable to apply PodDisruptionBudget changes: %v", err)
-						return err
-					}
+		pdbObj, err := c.pdbLister.PodDisruptionBudgets(pdb.Namespace).Get(pdb.Name)
+		if err == nil {
+			if pdbObj.Spec.MinAvailable != pdb.Spec.MinAvailable {
+				_, _, err = resourceapply.ApplyPodDisruptionBudget(ctx, c.pdbGetter, syncCtx.Recorder(), pdb)
+				if err != nil {
+					klog.Errorf("Unable to apply PodDisruptionBudget changes: %v", err)
+					return fmt.Errorf("Unable to apply PodDisruptionBudget changes: %v", err)
 				}
-				break
 			}
+		} else if errors.IsNotFound(err) {
+			_, _, err = resourceapply.ApplyPodDisruptionBudget(ctx, c.pdbGetter, syncCtx.Recorder(), pdb)
```

Comment:
is this the first time we actually create the a pdb? does it mean we previously didn't create it ?  (we didn't handle `IsNotFound`)

Reply:
This is the first time the pdb is created. Previously, there was a loop which assumed the pdb already exists. The case of creating the actual pdb was omitted.